gems/pending/OpenStackExtract/MiqOpenStackVm/MiqOpenStackInstance.rb

```diff
@@ -30,7 +30,7 @@ def instance
   end
 
   def snapshot_metadata
-    @snapshot_metadata ||= instance.metadata.get(:miq_snapshot)
+    @snapshot_metadata ||= instance.metadata.length > 0 && instance.metadata.get(:miq_snapshot)
```

Comment:
Yes, it cannot ensure :miq_snapshot metadata item exists, but it do avoid poping up the  FOG error message. I'll work on the better rescue way.

[----] E, [2015-10-21T10:25:54.691843 #21702:127fe9c] ERROR -- : Q-task_id([96fe7d66-77ff-11e5-ae20-000c29313f21]) excon.error     #<Excon::Errors::NotFound: Expected([200, 203]) <=> Actual(404 Not Found)
excon.error.response
  :body          => "{\"itemNotFound\": {\"message\": \"Metadata item was not found\", \"code\": 404}}"
  :headers       => {
    "Content-Length"       => "73"
    "Content-Type"         => "application/json; charset=UTF-8"
    "Date"                 => "Wed, 21 Oct 2015 14:07:20 GMT"
    "
   .......